### ISSUE TITLE
Snapshot failing because of nc

### ISSUE THREAD

**Issue by jodybro** [kubedb/project#400]
This block inside the `postgres-tools.sh` script blocks the snapshot object from completing its functions.

    #while ! nc "$DB_HOST" "$DB_PORT" -w 30 >/dev/null; do
    #  echo "Waiting... database is not ready yet"
    #  sleep 5
    # done

The db container is reachable at all three of the following addresses from the snapshot container but doesnt work when using `nc`:

- `db` or whatever the name of the db is
- `db.${namespace}.svc.cluster.local`
- IP address

Commentting it out and re-running the command that's generated by the container results in the dump to s3 (in my case) working like a charm.

This blocks the snapshot feature entirely for the time being

**Comment by the-redback** [comment]
Can you also provide more information about cluster so that we can reproduce? How you provision your cluster? Kubernetes version?

**Comment by jodybro** [comment]
The cluster was spun up via `kops` in a private DNS zone:

    $ kops create  cluster --name=${NAME} \
    --master-count=1 \
    --master-size=m4.large \
    --zones=ca-central-1a,ca-central-1b \
    --node-count=2 \
    --node-size=t3.medium \
    --dns-zone=${ZONE} \
    --state=${KOPS_STATE_STORE} \
    --dns private \
    --output yaml \
    --yes \
    --networking flannel

Running `v.1.11`

**Comment by jodybro** [comment]
Ill try upgrading my k8s version to see if that helps

**Comment by Edke** [comment]
Also not working for me, Kubernetes v.1.14.3.

@jodybro Any idea for workaround till fixed ? Can I override which image is used by Operator and fix ?

**Comment by kesselborn** [comment]
Hey: the problem occurs if you name your database `db`. The reason for this is that Kubernetes automatically creates env variables for services, among others an env var which is called `<SERVICE_NAME>_PORT`. 
If you name your database `db`, it'll create an env variable called `DB_PORT=tcp://<service-ip>:5432`. The backup script expects `DB_PORT` to just be the port number though

tl;dr: don't assign the name `db` to your postgres db.

**Comment by pavel-khritonenko** [comment]
The same issue with `MongoDB`. Is there any workaround except copying database with new name?

**Comment by keithlayne** [comment]
I just figured out what @kesselborn said before arriving here. I don't really see a workaround.

This is pretty painful - looks like you could use the service template to change the service port, but I'm pretty sure that would break backups as well. I don't see a way to actually specify the port through an arg or somehow supplying the backup pod with an env var.

Therefore...`DB_PORT` AFAICT is completely useless, and backup will only ever work if it falls back to the default.  Please correct me if I'm wrong here.

C'mon.  Validate your inputs. If you want me to believe you know what you're  doing with k8s, don't use scripts with unvalidated env vars called `<combo of uppercase letters and underscores>_PORT`.

**Comment by tamalsaha** [comment]
We don't use the Snapshot approach anymore. Obsolete.
